src/Common/ZooKeeper/ZooKeeperCachingGetter.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+#include <Common/ZooKeeper/Common.h>
+
+
+namespace zkutil
+{
+
+class ZooKeeperCachingGetter : boost::noncopyable
```

Comment:
`CacheBase` functionality is definitely not needed here. perhaps the word caching confused you. What it means here is that it returns previously saved session if it is not expired or recreated the new one otherwise.